lib/metadata/bucketclient/backend.js

```diff
@@ -27,75 +35,144 @@ class BucketClientInterface {
             });
             this.client = new bucketclient.RESTClient(bootstrap, log);
         }
+        this.init();
+    }
+
+    init() {
+        // get metadata information
+        const beginPath = '/default/metadataInformation/foo';
+        const log = logger.newRequestLogger();
+        this.client.request('GET', beginPath, log, null, null, (err, data) => {
+            if (!err) {
+                try {
+                    const info = JSON.parse(data);
+                    this.metadataVersion = Number(info.metadataVersion);
+                    this.extendedVersion =
+                        this.metadataVersion > METADATA_DEFAULT_VERSION;
+                } catch (error) {
+                    log.error('error parsing metadata information', { error });
+                }
+            } else if (!err.RouteNotFound) {
```

Comment:
It will go to the next step updating the `gotMetadataInfo` without changing the `metadataVersion`. In this case it means metadata does not support the `metadataInformation` query thus does not support versioning.